### ISSUE TITLE
[Bug report] hamiltonian_for_qutip_dynamics Hamiltonian expansion order

### ISSUE THREAD

**Issue by zachparrott** [scqubits/scqubits#247]
### Documentation check
If the bug manifests in unexpected behavior (as opposed to a crash), confirm that you have consulted the 
[API documentation](https://scqubits.readthedocs.io/en/latest/api-doc/apidoc.html)

 - [x] I have checked the API documentation.
 - [x] I could not locate relevant information in the documentation or information is missing. 

### Describe the bug
In the documentation for `hamiltonian_for_qutip_dynamics` it says in the argument `free_var_func_dict` that you can specify the order to which the Hamiltonian is expanded about such as:
```
def flux_t(t, args):
            return np.sin(t*2)
free_var_func_dict = {"Φ1": (flux_t, 2)}
```

 but if I do so I get the following error in the `parameter_func` method:

```
File c:\users\zlp\github\scqubits\scqubits\core\circuit_routines.py:2577, in CircuitRoutines.hamiltonian_for_qutip_dynamics.<locals>.parameter_func(t, args, parameter_expr, self, free_var_func_dict, lambdify_func)
   [2568] def parameter_func(
   [2569]     t,
   [2570]     args,
   (...)
   [2574]     lambdify_func=lambdify_func,
   [2575] ):
   [2576]     return lambdify_func(
-> [2577]         *[
   [2578]             free_var_func_dict[sym.name](t, args)
   [2579]             for sym in parameter_expr.free_symbols
   [2580]         ]
   [2581]     )

TypeError: 'tuple' object is not callable
```
Issues persists with the order set at anything. 

In the documentation page [Custom Circuits / Parametric Driving](https://scqubits.readthedocs.io/en/v4.2/guide/circuit/ipynb/custom_circuit_parametric_driving.html) it does not use the format shown in the docstrings and just passes the time dependent function for the given variable. 

Reading through the code for `hamiltonian_for_qutip_dynamics` it is not clear to me whether anything prior to `parameter_func` is using the order if provided. Is this an unimplemented functionality? 

Otherwise I really appreciate this functionality as it presently works!

### Expected behavior
QuTiP's sesolve would successfully run. 

### To Reproduce
If not clear from your description above, please provide the steps to reproduce the behavior:

### OS and version used (please complete the following information):
 - OS: [Windows]
 - scqubits version [4.2]
 - Python version [3.9]

### Additional context
Any additional information you would like to provide to aid us.

**Comment by saipavanc** [comment]
Hi Zach, thanks for raising the issue. I just discussed with Rohan in my group yesterday, I initially implemented it with the order-by-order expansion of the time dependent variable. But later decided against it, and forgot to update the docstring. The function in the current form (with no order expansion) is an implemented functionality, I will push changes to fix them soon. 

I am very glad that you find this functionality useful!

**Comment by saipavanc** [comment]
Fixed it in the commit 656b907

**Comment by zachparrott** [comment]
Thanks Sai. Can you elaborate how the current implementation actually works? It was not totally obvious reading through the source code.
